tests/src/test/java/org/eclipse/hono/tests/registry/CredentialsHttpIT.java

```diff
@@ -403,9 +403,11 @@ public void testGetAddedCredentials(final VertxTestContext context) {
                 HttpURLConnection.HTTP_NO_CONTENT)
                 .compose(ar -> registry.getCredentials(TENANT, deviceId))
                 .setHandler(context.succeeding(b -> {
-                    context.verify(() -> assertThat(b.toJsonArray())
-                            .isEqualTo(new JsonArray().add(JsonObject.mapFrom(hashedPasswordCredential))));
-                    context.completeNow();
+                    final PasswordCredential cred = b.toJsonArray().getJsonObject(0).mapTo(PasswordCredential.class);
+                    cred.getSecrets().forEach(secret -> {
+                        context.verify(() -> assertThat(secret.getId()).isNotNull());
+                        context.completeNow();;
```

Comment:
you shouldn't complete the context for each secret ...

Reply:
Good catch, thanks